### PR TITLE
specify the cuda arch name and bin to decrease the compile time for i…

### PR DESCRIPTION
**Pull request by wopeizl**
…nference test=develop
fix https://github.com/PaddlePaddle/Paddle/pull/16901#pullrequestreview-229001528

### CODE REVIEW

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -753,7 +755,8 @@ EOF
     if [[ "$1" != "" ]]; then
       parallel_number=$1
     fi
-    cmake .. -DWITH_DISTRIBUTE=OFF -DON_INFER=ON
+    cmake .. -DWITH_DISTRIBUTE=OFF -DON_INFER=ON -DCUDA_ARCH_NAME=${CUDA_ARCH_NAME:-All} -DCUDA_ARCH_BIN=${CUDA_ARCH_BIN}
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: 这里为什么用All呢？应该用Auto

Comment (wopeizl) [reply paddle/scripts/paddle_build.sh]: 谢谢指出

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -677,24 +627,24 @@ function card_test() {
...
 set +x
         EXIT_CODE=0;
-        test_cases=$(ctest -N -V)
-        exclusive_tests=''
-        single_card_tests=''
-        multiple_card_tests=''
-        is_exclusive=''
-        is_multicard=''
+        test_cases=$(ctest -N -V) # get all test cases
+        exclusive_tests=''      # cases list which would been run exclusively
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: would be

File: paddle/scripts/paddle_build.sh [hunk 3]
```diff
@@ -677,24 +627,24 @@ function card_test() {
...
+        single_card_tests=''    # cases list which would take one graph card
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: one graph card-> one GPU

File: paddle/scripts/paddle_build.sh [hunk 4]
```diff
@@ -677,24 +627,24 @@ function card_test() {
...
+        multiple_card_tests=''  # cases list which would take one multiple card, most cases would be two cards
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: 这句话不通顺。one multiple card是指？two cards->two GPUs

File: paddle/scripts/paddle_build.sh [hunk 5]
```diff
@@ -740,9 +690,9 @@ set +x
                 testcase=''
         done <<< "$test_cases";
 
-        card_test "$single_card_tests" 1
-        card_test "$multiple_card_tests" 2
-        card_test "$exclusive_tests"
+        card_test "$single_card_tests" 1    # run cases with single card
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: single card->single GPU。单卡翻译成英文是single GPU。全文同。

**Review by luotao1 — COMMENTED** [review]
https://github.com/PaddlePaddle/Paddle/pull/16901#discussion_r277255132
那SERIAL还需要写么？

**Comment by wopeizl** [comment]
> [#16901 (comment)](https://github.com/PaddlePaddle/Paddle/pull/16901#discussion_r277255132)
> 那SERIAL还需要写么？

不用了

**Comment by luotao1** [comment]
那原来写SERIAL的地方统一改一下吧。

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**